.github/workflows/ci-app_cloud_e2e_test.yml

```diff
@@ -0,0 +1,186 @@
+name: cloud-testing
+
+# Used to run the e2e tests on lightning.ai
+
+on: # Trigger the workflow on push or pull request, but only for the master branch
+  push:
+    branches:
+      - "master"
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+jobs:
+  # This is job should once only once per PR to detect file changes so run required jobs.
+  # see .github/file-filters.yml to define file filters and run the jobs based on the output of each filter.
+  # More info: https://github.com/marketplace/actions/paths-changes-filter
+
+  changes:
+    runs-on: ubuntu-latest
+    # Set job outputs to the values from filter step
+    outputs:
+      app_examples: ${{ steps.filter.outputs.app_examples }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: .github/file-filters.yml
+
+  cloud-test:
+    name: Cloud Test
+    needs: changes
+    if: ${{ needs.changes.outputs.app_examples == 'true' }}
```

Comment:
Are we going to mark `could-test` as one of the required jobs before merge? If so, we need to run `cloud-test` regardless of the paths that changes are made to because when a PR has no changes in the paths, required jobs will be skipped and thus won't satisfy the requirement for merging the PR.

If GitHub provided the feature "conditional required checks" in the thread below, it would've been much easier.
- https://github.community/t/feature-request-conditional-required-checks/16761
- https://github.com/orgs/github-community/discussions/13690

---

Anyway, my comment won't apply here if we don't mark this job as required.

Reply:
well, yes, there has to be a bigger update on running and enabling required checks, so let's leave this as a quick solution and continue in other PR :rabbit:

Reply:
Sure, we can merge this, but still I don't see this as a solution to anything. There isn't any difference between this manual path splitter and the `paths` GitHub already provides, is there?
```yaml
on:
  pull_request:
    paths:
      - "src/lightning_app/**"
      - "tests/tests_app_examples/**"
      - "requirements/app/**"
      - "examples/app_*"
      - "setup.py"
      - "src/pytorch_lightning/__version__.py"
```

---

But yes, we can improve on this later as this PR's scope is to add the cloud test!

Reply:
True. We can do that in another PR.

I found this that might be useful: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks

Reply:
that looks interesting, just curious about the blind zone, when this always passes job will be returning green, but the true job will be hanging in ques, then it is usually not listed among checks... so it may happen that it'll be merged?

Reply:
I want to test it, but I guess when having two workflows with the same and one of them is not skipped, it should wait for success on both.